sources2csr/derived_values.py

```diff
@@ -0,0 +1,41 @@
+from collections import defaultdict
+from operator import attrgetter
+from typing import List, Dict
+
+from dateutil.relativedelta import relativedelta
+
+from csr.csr import CentralSubjectRegistry, Diagnosis
+
+
+def add_derived_values(subject_registry: CentralSubjectRegistry) -> CentralSubjectRegistry:
+    """
+
+    :param subject_registry:
+    :return:
+    """
+
+    # Collect relevant diagnosis aggregates per individual
+    diagnoses_per_individual: Dict[str, List[Diagnosis]] = defaultdict(list)
+    for diagnosis in subject_registry.diagnoses:
+        diagnoses_per_individual[diagnosis.individual_id].append(diagnosis)
+    diagnosis_count_per_individual = {}
+    first_diagnosis_date_per_individual = {}
+    for individual_id, diagnoses in diagnoses_per_individual.items():
+        diagnosis_count = len(diagnoses)
+        diagnosis_count_per_individual[individual_id] = diagnosis_count
+        if diagnosis_count > 0:
+            first_diagnosis = sorted(diagnoses, key=attrgetter('diagnosis_date'))[0]
```

Comment:
This fails if there are more than one diagnosis without diagnosis_date with exception:
`TypeError: '<' not supported between instances of 'NoneType' and 'NoneType'`

Test case I tried (modified test_derived_values.py): 
```
@pytest.fixture
def registry_with_missing_diagnosis_data() -> CentralSubjectRegistry:
    individuals = [
        Individual(individual_id='P4')
    ]
    diagnoses = [
        Diagnosis(diagnosis_id='D4.1', individual_id='P4'),
        Diagnosis(diagnosis_id='D4.2', individual_id='P4')
    ]
    return CentralSubjectRegistry(individuals=individuals, diagnoses=diagnoses)


def test_diagnosis_aggregates_with_missing_data(registry_with_missing_diagnosis_data):
    subject_registry = add_derived_values(registry_with_missing_diagnosis_data)
    expected_counts = {'P4': 2}
    expected_age = {'P4': None}
    for individual in subject_registry.individuals:
        assert individual.age_first_diagnosis == expected_age[individual.individual_id]
        assert individual.diagnosis_count == expected_counts[individual.individual_id]
```

Reply:
Nice catch! Also, difficult to think what should be considered the 'first' diagnosis if they don't have a date :). But I will filter on having a date field first then. Thanks for adding the test case.

Reply:
Has been fixed now.